### PR TITLE
Hook Up Daily Build to All E2Es Plus More SPEEEEED!

### PR DESCRIPTION
**Pull request by davidvonthenen**
## What this PR does / why we need it

This PR introduces the ability to download the generated nightly build and trigger all E2Es to test said build.

The full list of changes includes:
- changing the cron time to 1am pacific (the times are in UTC which is why it's "0 9 * * *"
- pass the "date" of the daily build to each E2E... that in turn gets passed to the `download-or-build-tce.sh` script. If the DAILY_BUILD value is there, we will download... otherwise we will build ONLY the platform we are running on (which is usually "linux-amd64" but hey... this is dynamic) and saves us from building 3 other platforms
- just a bunch of clean up

## Which issue(s) this PR fixes

NA

## Describe testing done for PR

NA

## Special notes for your reviewer

NA

### CODE REVIEW

File: .github/workflows/e2e-azure-management-and-workload-cluster.yaml
Comment (dvonthenen) [reply .github/workflows/e2e-azure-management-and-workload-cluster.yaml:65]: I don't believe it should... hahah! The daily build event is the only thing that populates that ENV value. If this doesn't work like I think it would, then we can always revert this.

**Comment by dvonthenen** [comment]
I simulated this workflow here:
https://github.com/dvonthenen/ghactiontest/actions

I ran what would normally be an e2e run here (note the daily-build value is empty):
https://github.com/dvonthenen/ghactiontest/runs/5792041558?check_suite_focus=true

Then I did a manual trigger of a "daily build", and you can see the value being populated in this run:
https://github.com/dvonthenen/ghactiontest/runs/5792064173?check_suite_focus=true

I think we are good to go with this PR!

**Review by stmcginnis — DISMISSED** [review]
Looks like everything is happy. We can keep an eye on daily build jobs after this merges and react to that if needed.

**Review by stmcginnis — APPROVED** [review]

**Review by jpmcb — APPROVED** [review]

**Comment by dvonthenen** [comment]
Merging since we have the 2 approvers

**Pull request merged**
